litellm/caching.py

```diff
@@ -1217,6 +1217,410 @@ async def async_get_cache(self, key, **kwargs):
     async def _index_info(self):
         return await self.index.ainfo()
 
+class QdrantSemanticCache(BaseCache):
+    def __init__(
+            self, 
+            qdrant_url=None,
+            qdrant_api_key = None,
+            collection_name=None, 
+            similarity_threshold=None,
+            quantization_config=None,
+            embedding_model="text-embedding-ada-002",
+            host_type = None
+            ):
+        from litellm.llms.custom_httpx.http_handler import (
+            _get_httpx_client, 
+            _get_async_httpx_client
+        )
+
+        if collection_name is None:
+            raise Exception("collection_name must be provided, passed None")
+        
+        self.collection_name = collection_name
+        print_verbose(
+            f"qdrant semantic-cache initializing COLLECTION - {self.collection_name}"
+        )
+
+        if similarity_threshold is None:
+            raise Exception("similarity_threshold must be provided, passed None")
+        self.similarity_threshold = similarity_threshold
+        self.embedding_model = embedding_model
+
+        if host_type=="cloud":
```

Comment:
why do i need a 'host_type'? 

is setting base url not sufficient?

Reply:
Base URL works as well. Will update this part soon!